lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -454,15 +452,26 @@ putAddress addr@(Address bs)
         ++ ": " ++ show addr
   where l = BS.length bs
 
-singleAddressFromKey :: forall n. KnownNetwork n => Proxy n -> XPub -> Address
-singleAddressFromKey _ xpub = Address $ BL.toStrict $ runPut $ do
+singleAddressFromKey :: forall n. KnownNetwork n => Proxy n -> ByteString -> Address
```

Comment:
The type was better before using the `XPub`. What's the rationale for choosing a raw ByteString instead?

Reply:
We don't have a `ChainCode`.

Reply:
Yes, but we can extract the relevant bits from within `singleAddressFromKey`. For lack of a better intermediate type, `XPub` is cleaner than a raw `ByteString`